packages/esm-patient-medications-app/src/order-basket/order-basket-search/order-basket-search-results.component.tsx

```diff
@@ -13,16 +13,29 @@ export interface OrderBasketSearchResultsProps {
   searchTerm: string;
   setSearchTerm: (value: string) => void;
   onSearchResultClicked: (searchResult: OrderBasketItem, directlyAddToBasket: boolean) => void;
+  searchInputRef: RefObject<HTMLInputElement>;
 }
 
 export default function OrderBasketSearchResults({
+  searchInputRef,
   searchTerm,
   setSearchTerm,
   onSearchResultClicked,
 }: OrderBasketSearchResultsProps) {
   const { t } = useTranslation();
   const isTablet = useLayoutType() === 'tablet';
   const { drugs, isLoading, error } = useDrugSearch(searchTerm);
+  const [isSearchIputCleared, setIsSearchInputCleared] = useState(false);
+  function handleClick() {
+    setIsSearchInputCleared(true);
+    setSearchTerm('');
+  }
+  useEffect(() => {
+    if (isSearchIputCleared && searchInputRef.current) {
+      searchInputRef.current.focus();
+      setIsSearchInputCleared(false);
+    }
```

Comment:
This code block can be moved to the handleClick function above instead of setting a state and then running the useEffect